src/test/java/com/vinylteam/vinyl/service/impl/DefaultOfferServiceTest.java

```diff
@@ -158,4 +161,49 @@ void getListOfShopsIdsWithNullOffersListTest() {
         assertTrue(actualIds.isEmpty());
     }
 
+    @Test
+    @DisplayName("Checks that offers are merged during dynamic update")
+    void mergeOfferChangesTest() {
+        //when
+        VinylParser parser = new CloneNlParser();
+        Offer oldOffer = dataGenerator.getOfferWithUniqueVinylIdAndShopId(12, 4);
+        RawOffer newlyLoadedOffer = dataGenerator.getRawOfferWithShopIdAndNumber(4, 12);
+        var newPrice = newlyLoadedOffer.getPrice() - 2d;
+        newlyLoadedOffer.setPrice(newPrice);
+        assertNotEquals(newlyLoadedOffer.getPrice(), oldOffer.getPrice());
+        offerService.mergeOfferChanges(oldOffer, parser, newlyLoadedOffer);
```

Comment:
You could have used:
expectedOffer, actualOffer(oldOffer before and after merging).
And then assertEquals(expectedOffer, actualOffer);
I appreciate that you used dataGenerator,

Reply:
these offers will not be equal, do different generator methods guarantee it? And the merge method does not set all fields, only price, currency, inStock

Reply:
made attempt to use more informative names